File: Fevi/src/main/java/com/app/fevir/movie/list/adapter/MovieListAdapter.java
package com.app.fevir.movie.list.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.app.fevir.movie.list.domain.Card;
import com.app.fevir.util.picaso.CircleTransform;
import com.app.fevir.views.listener.RecyclerItemClickListener;
import com.squareup.picasso.Picasso;
import com.vikicast.app.R;

import java.util.ArrayList;
import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;

/**
 * Created by 1000742 on 15. 1. 5..
 */
public class MovieListAdapter extends RecyclerView.Adapter<MovieListAdapter.CardHolder> {

    private Context context;
    private List<Card> cardList;
    private RecyclerItemClickListener recyclerItemClickListener;

    public MovieListAdapter(Context context) {
        super();
        this.context = context;
        cardList = new ArrayList<>();
    }

    public void add(Card card) {
        cardList.add(card);
    }

    public Card getItem(int position) {
        return cardList.get(position);
    }

    @Override
    public CardHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(context).inflate(R.layout.fragment_facebook, parent, false);
        CardHolder adapterHolder = new CardHolder(v);

        ButterKnife.bind(adapterHolder, v);

        return adapterHolder;
    }

    @Override
    public void onBindViewHolder(CardHolder holder, int position) {

        Card card = getItem(position);

        holder.name.setText(card.getName());
        holder.description.setText(card.getDescription());
        holder.time.setText(card.getUpdatedTime());

        Picasso.with(context).load(card.getProfileImage()).transform(new CircleTransform()).into(holder.profile);
        Picasso.with(context).load(card.getPicture()).into(holder.picture);

        holder.faList.setOnClickListener(v -> {
            if (recyclerItemClickListener != null) {
                recyclerItemClickListener.onItemClick(MovieListAdapter.this, position);
            }
        });
    }

    @Override
    public int getItemCount() {
        return cardList.size();
    }

    public void setRecyclerItemClickListener(RecyclerItemClickListener recyclerItemClickListener) {
        this.recyclerItemClickListener = recyclerItemClickListener;
    }

    static class CardHolder extends RecyclerView.ViewHolder {
        @Bind(R.id.fa_name)
        TextView name;
        @Bind(R.id.fa_profile)
        ImageView profile;
        @Bind(R.id.fa_picture)
        ImageView picture;
        @Bind(R.id.fa_description)
        TextView description;
        @Bind(R.id.fa_time)
        TextView time;
        @Bind(R.id.fa_list)
        LinearLayout faList;

        public CardHolder(View itemView) {
            super(itemView);
        }
    }

}

File: Fevi/src/main/java/com/app/fevir/movie/list/adapter/MovieListAdapter.kt
package com.app.fevir.movie.list.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import butterknife.Bind
import butterknife.ButterKnife
import com.app.fevir.R
import com.app.fevir.movie.list.domain.Card
import com.app.fevir.util.picaso.CircleTransform
import com.app.fevir.views.listener.RecyclerItemClickListener
import com.squareup.picasso.Picasso
import java.util.*

/**
 * Created by 1000742 on 15. 1. 5..
 */
internal class MovieListAdapter(private val context: Context) : RecyclerView.Adapter<MovieListAdapter.CardHolder>() {
    lateinit  var cardList: MutableList<Card>
    private var recyclerItemClickListener: RecyclerItemClickListener? = null

    init {
        cardList = ArrayList()
    }

    fun add(card: Card) {
        cardList.add(card)
    }

    fun getItem(position: Int): Card = cardList.get(position)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardHolder {
        val v = LayoutInflater.from(context).inflate(R.layout.fragment_facebook, parent, false)
        val adapterHolder = CardHolder(v)

        ButterKnife.bind(adapterHolder, v)

        return adapterHolder
    }

    override fun onBindViewHolder(holder: CardHolder, position: Int) {

        val card = getItem(position)

        holder.name.text = card.name
        holder.description.text = card.description
        holder.time.text = card.updatedTime

        Picasso.with(context).load(card.profileImage).transform(CircleTransform()).into(holder.profile)
        Picasso.with(context).load(card.picture).into(holder.picture)

        holder.faList.setOnClickListener { v ->
            if (recyclerItemClickListener != null) {
                recyclerItemClickListener!!.onItemClick(this@MovieListAdapter as RecyclerView.Adapter<RecyclerView.ViewHolder>, position)
            }
        }
    }

    override fun getItemCount(): Int = cardList.size

    fun setRecyclerItemClickListener(recyclerItemClickListener: RecyclerItemClickListener) {
        this.recyclerItemClickListener = recyclerItemClickListener
    }

    internal class CardHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        @Bind(R.id.fa_name)
        lateinit var name: TextView
        @Bind(R.id.fa_profile)
        lateinit var profile: ImageView
        @Bind(R.id.fa_picture)
        lateinit var picture: ImageView
        @Bind(R.id.fa_description)
        lateinit var description: TextView
        @Bind(R.id.fa_time)
        lateinit var time: TextView
        @Bind(R.id.fa_list)
        lateinit var faList: LinearLayout
    }

}
